src/server/controllers/user.js

```diff
@@ -1,39 +1,63 @@
-import bcrypt from 'bcrypt';
-import jwt from 'jsonwebtoken';
-import { PrismaClient } from '@prisma/client'
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import { PrismaClient } from "@prisma/client";
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const jwtSecret = "mysecret";
 
 const register = async (req, res) => {
-    const { username, password } = req.body;
-
-    const createdUser = null;
-
-    res.json({ data: createdUser });
+  const { username, password } = req.body;
+  const saltRounds = 10;
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+
+  const existingUser = await prisma.user.findUnique({
+    where: { username },
+  });
+
+  if (existingUser) {
+    return res.status(400).json({ message: "User already exists" });
+  }
+
+  const newUser = await prisma.user.create({
+    data: {
+      username,
+      password: hashedPassword,
+    },
+  });
+
+  const token = jwt.sign({ userId: newUser.id }, jwtSecret, {
+    expiresIn: "24h",
+  });
+
+  res.status(201).json({ data: newUser, token });
 };
 
 const login = async (req, res) => {
-    const { username, password } = req.body;
+  const { username, password } = req.body;
 
-    const foundUser = null;
+  try {
+    const foundUser = await prisma.user.findUnique({
+      where: { username },
+    });
 
     if (!foundUser) {
-        return res.status(401).json({ error: 'Invalid username or password.' });
+      return res.status(401).json({ error: "Invalid username or password." });
     }
 
-    const passwordsMatch = false;
+    const passwordsMatch = await bcrypt.compare(password, foundUser.password);
 
-    if (!passwordsMatch) {
-        return res.status(401).json({ error: 'Invalid username or password.' });
+    if (passwordsMatch) {
+      const token = jwt.sign({ username: foundUser.username }, jwtSecret, {
+        expiresIn: "24h",
```

Comment:
expiresIn is also a good candidate for an environment variable so that we don't have to change code to change how long a token exists for.